File: app/src/main/java/com/goranch/publicapis/ui/food/IDataRepository.java
package com.goranch.publicapis.ui.food;

import com.goranch.publicapis.api.model.food.nutrition.Hit;
import com.goranch.publicapis.api.model.food.recipe.Recipe;

import java.util.List;

import io.reactivex.Observable;


public interface IDataRepository {
    Observable<List<Recipe>> searchRecipes(String searchQuery, String apiKey);

    Observable<Recipe> getRecipe(String recipeId, String apiKey);

    Observable<List<Hit>> getNaturalLanguageNutritionInfo(String query);
}

File: app/src/main/java/com/goranch/publicapis/ui/food/repository/IDataRepository.kt
package com.goranch.publicapis.ui.food.repository

import com.goranch.publicapis.api.model.food.nutrition.Food
import com.goranch.publicapis.api.model.food.recipe.Recipe

import io.reactivex.Observable


interface IDataRepository {
    fun searchRecipes(searchQuery: String, apiKey: String): Observable<List<Recipe>>

    fun getRecipe(recipeId: String, apiKey: String): Observable<Recipe>

    fun getNaturalLanguageNutritionInfo(query: String): Observable<List<Food>>
}
